### ISSUE TITLE
Cross platform analytics

### ISSUE THREAD

**Issue by aidanbdh** [aidanbdh/Group-Project-Ideas#1]
Collect analytics from user's:
> Website
> Slack channel
> GitHub project?
> App

Collect data:
> Views
> Mentions
> Keywords
> Revisits
> Time spent

Interface:
> Website
> Mobile app
> Email
> Messaging
> Slack messages
> Secure login

Whats your opinion?

**Comment by ansonhwo** [comment]
If we're gonna do analytics, then it would be a lot more interesting if we did it for something that can be largely scaled. I feel like Slack or Github would be too narrow in scope for this type of idea. 

In related news, I found an API called Inspectlet [https://www.inspectlet.com/docs] that tracks and records user interactions on websites, and also plots heatmaps of mouse & scrolling activity. I'm not too familiar with collecting this type of data, but maybe it would be interesting to try and practice implementing the types of features that Inspectlet has.

**Comment by aidanbdh** [comment]
Good idea! Do you have any ideas for what we should develop it for for larger scaling?

Also, if we are going to design it for large scale, then we need to address the issue of database size. The heroku databases have a limited number of size and connections unless you want to pay more. So we can host our own db, do something where we just give the users the data to use how they want or something else.

**Comment by ansonhwo** [comment]
At the moment, not right now. I'm going to have to do more reading about this sort of thing. But you're right about the DB size issue; I think it would be cool to have some kind of customizable dashboard or something where the user can choose which data they care about most, so that they can check those first at a glance.
